python/variable.cpp

```diff
@@ -62,34 +62,43 @@ void bind_init_0D_native_python_types(py::class_<Variable> &c) {
 }
 
 void bind_init_0D_numpy_types(py::class_<Variable> &c) {
-  c.def(py::init([](py::buffer &b, const std::optional<py::buffer> &v,
-                    const units::Unit &unit, py::object &dtype) {
-          py::buffer_info info = b.request();
-          if (info.ndim == 0) {
-            auto arr = py::array(b);
-            auto varr = v ? std::optional{py::array(*v)} : std::nullopt;
-            return doMakeVariable({}, arr, varr, unit, dtype);
-          } else if (info.ndim == 1 &&
-                     scipp_dtype(dtype) == core::dtype<Eigen::Vector3d>) {
-            return do_init_0D<Eigen::Vector3d>(
-                b.cast<Eigen::Vector3d>(),
-                v ? std::optional(v->cast<Eigen::Vector3d>()) : std::nullopt,
-                unit);
-          } else if (info.ndim == 1 &&
-                     scipp_dtype(dtype) == core::dtype<Eigen::Quaterniond>) {
-            return do_init_0D<Eigen::Quaterniond>(
-                Eigen::Quaterniond(b.cast<std::vector<double>>().data()),
-                v ? std::optional(Eigen::Quaterniond(
-                        v->cast<std::vector<double>>().data()))
-                  : std::nullopt,
-                unit);
-          } else {
-            throw scipp::except::VariableError(
-                "Wrong overload for making 0D variable.");
-          }
-        }),
-        py::arg("value").noconvert(), py::arg("variance") = std::nullopt,
-        py::arg("unit") = units::one, py::arg("dtype") = py::none());
+  c.def(
+      py::init([](py::buffer &b, const std::optional<py::buffer> &v,
+                  const units::Unit &unit, py::object &dtype) {
+        py::buffer_info info = b.request();
+        if (info.ndim == 0) {
+          auto arr = py::array(b);
+          auto varr = v ? std::optional{py::array(*v)} : std::nullopt;
+          return doMakeVariable({}, arr, varr, unit, dtype);
+        } else if (info.ndim == 1 &&
+                   scipp_dtype(dtype) == core::dtype<Eigen::Vector3d>) {
+          return do_init_0D<Eigen::Vector3d>(
+              b.cast<Eigen::Vector3d>(),
+              v ? std::optional(v->cast<Eigen::Vector3d>()) : std::nullopt,
+              unit);
+        } else if (info.ndim == 1 &&
+                   scipp_dtype(dtype) == core::dtype<scipp::core::time_point>) {
+          return do_init_0D<scipp::core::time_point>(
+              b.cast<scipp::core::time_point>(),
+              v ? std::optional(v->cast<scipp::core::time_point>())
+                : std::nullopt,
+              unit);
+
+        } else if (info.ndim == 1 &&
+                   scipp_dtype(dtype) == core::dtype<Eigen::Quaterniond>) {
+          return do_init_0D<Eigen::Quaterniond>(
+              Eigen::Quaterniond(b.cast<std::vector<double>>().data()),
```

Comment:
Quaternions have been removed from scipp a while ago, you still have quite a few merge conflicts, I believe. Consider not included all the Python changes in this PR to avoid too many conflicts.